keys.go

```diff
@@ -0,0 +1,63 @@
+package trails
+
+import "sort"
+
+type Key string
+
+const (
+	// AppPropsKey stashes additional props to be included in HTTP responses.
+	AppPropsKey Key = "AppPropsKey"
+
+	// CurrentUserKey stashes the currentUser for a session.
+	CurrentUserKey Key = "CurrentUserKey"
+
+	// IpAddrKey stashes the IP address of an HTTP request being handled by trails.
+	IpAddrKey Key = "IpAddrKey"
+
+	// RequestIDKey stashes a unique UUID for each HTTP request.
+	RequestIDKey Key = "RequestIDKey"
+
+	// SessionKey stashes the session associated with an HTTP request.
+	SessionKey Key = "SessionKey"
+)
+
+// String formats the stringified key with additional contextual information
+func (k Key) String() string {
+	return "trails context key: " + string(k)
+}
+
+type ByKey []Key
+
+var _ sort.Interface = ByKey([]Key{})
+
+func (k ByKey) Len() int           { return len(k) }
+func (k ByKey) Swap(i, j int)      { k[i], k[j] = k[j], k[i] }
+func (k ByKey) Less(i, j int) bool { return string(k[i]) < string(k[j]) }
+
+// UniqueSort sorts, uniques and removes zero keys.
+func (k ByKey) UniqueSort() ByKey {
```

Comment:
It looks like this is used in `WithCtxKeys`. I'm curious how this came to be? Is it trying to make sure this is as tidy as possible? Or is it preventing an issue you bumped into? Same with `type ByKey []Key` in general